client/src/test/java/io/pravega/client/stream/impl/ControllerImplTest.java

```diff
@@ -1194,4 +1253,138 @@ public void testCutpointSuccessors() throws Exception {
                                      new Segment(scope, stream, 6), new Segment(scope, stream, 7)),
                      successors);
     }
+
+    @Test
+    public void testSegmentsValid() throws Exception {
+        String scope = "scope1";
+        String stream = "stream1";
+        Stream s = new StreamImpl(scope, stream);
+
+        Map<Segment, Long> startSegments = new HashMap<>();
+        startSegments.put(new Segment(scope, stream, 0), 4L);
+        startSegments.put(new Segment(scope, stream, 1), 6L);
+        StreamCut cut = new StreamCutImpl(s, startSegments);
+
+        Map<Segment, Long> endSegments = new HashMap<>();
+        endSegments.put(new Segment(scope, stream, 6), 10L);
+        endSegments.put(new Segment(scope, stream, 7), 10L);
+        StreamCut endSC = new StreamCutImpl(s, endSegments);
+
+        Set<Segment> result = controllerClient.getSegments(cut, endSC).get().getSegments();
+        assertEquals(ImmutableSet.of(new Segment(scope, stream, 0), new Segment(scope, stream, 1),
+                new Segment(scope, stream, 2), new Segment(scope, stream, 3),
+                new Segment(scope, stream, 4), new Segment(scope, stream, 5),
+                new Segment(scope, stream, 6), new Segment(scope, stream, 7)),
+                result);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testSegmentsError() throws Exception {
+        String scope = "scope1";
+        String stream = "stream1";
+        Stream s = new StreamImpl(scope, stream);
+
+        Map<Segment, Long> startSegments = new HashMap<>();
+        startSegments.put(new Segment(scope, stream, 3), 4L);
+        startSegments.put(new Segment(scope, stream, 2), 4L);
+        startSegments.put(new Segment(scope, stream, 1), 6L);
+        StreamCut cut = new StreamCutImpl(s, startSegments);
+
+        Map<Segment, Long> endSegments = new HashMap<>();
+        endSegments.put(new Segment(scope, stream, 0), 10L);
+        endSegments.put(new Segment(scope, stream, 7), 10L);
+        StreamCut endSC = new StreamCutImpl(s, endSegments);
+
+        controllerClient.getSegments(cut, endSC).get().getSegments();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testSegmentsError2() throws Exception {
```

Comment:
Can we have something more descriptive than `...Error2`?

Reply:
renamed the test methods.